openassessment/tests/test_workflow_batch_update_api.py

```diff
@@ -0,0 +1,545 @@
+
+        pw_miles = PeerWorkflow.objects.get(student_id=miles["student_id"])
+        pw_miles.item_id = "item_2"
+        pw_miles.save()
+
+        blocked = update_api.get_blocked_peer_workflows(item_id="item_2")
+        # we expect 0 blocked submissions for "item_2"
+        self.assertEqual(len(blocked), 0)
+
+    def test_is_flexible_peer_grading_on(self):
+        workflow_requirements = {"peer": {
+            "must_grade": 2,
+            "must_be_graded_by": 3,
+            "enable_flexible_grading": True
+        }}
+        ora_block = MockOraBlock(workflow_requirements)
+        course_block = MockCourseBlock(False)
+        self.assertTrue(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        ora_block.requirements['peer']['enable_flexible_grading'] = False
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        ora_block.requirements['peer']['enable_flexible_grading'] = False
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        del (ora_block.requirements['peer']['enable_flexible_grading'])
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        del (ora_block.requirements['peer'])
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        course_block.force_on_flexible_peer_openassessments = True
+        self.assertTrue(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+    @patch('openassessment.workflow_batch_update_api.modulestore')
+    @patch('openassessment.workflow_batch_update_api.UsageKey.from_string')
+    def test_get_workflow_update_data(self, mocked_from_string, mocked_modulestore):
+        mocked_modulestore.return_value = MockModulestore()
+        mocked_from_string.side_effect = mock_from_string
+        #
+        peer_workflows = self.get_peer_workflows_for_test_get_workflow_update_data()
+        wup = update_api.get_workflow_update_data(peer_workflows)
+
+        #
+        self.assertEqual(len(wup["courses"]), 2)
+        self.assertEqual(wup["courses"][0]["course_id"], "course_id_1")
+        self.assertEqual(wup["courses"][0]["assessments"][0]["item_id"], "item_id_1")
+        self.assertEqual(wup["courses"][0]["assessments"][1]["item_id"], "item_id_2")
+        self.assertEqual(wup["courses"][1]["course_id"], "course_id_2")
+        self.assertEqual(wup["courses"][1]["assessments"][0]["item_id"], "item_id_3")
+
+    # pylint: disable=unused-variable
+    def get_peer_workflows_for_test_get_workflow_update_data(self):
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+        miles_sub, miles = self._create_student_and_submission("Miles", "Miles's answer")
+        pat_sub, pat = self._create_student_and_submission("Pat", "Pat's answer")
+
+        pw_tim = PeerWorkflow.objects.get(student_id=tim["student_id"])
```

Comment:
you could also modify create_student_and_submission to take these as optional params